### DISCUSSION TITLE
Websocket Client

### DISCUSSION THREAD

**Discussion by kamakazikamikaze** [Neoteroi/BlackSheep#232] [Ideas]
I found this project yesterday and am really excited to refactor some existing projects of mine to use this instead.

One thing that prevents me from doing so is the lack of a Websocket Client. Some other libraries, like Tornado and AIOHTTP, include this feature.

I see that the server-side implementation of Websockets was added not too long ago. Is there a plan to add a Websocket client?

**Comment by RobertoPrevato** [comment]
Hi @kamakazikamikaze
Thanks for your kind words. I took some time to reply to think about this. I never considered adding WebSocket support in the client implementation in BlackSheep, and I don't see this happening, because of lack of time maintaining so many features.
I am even tempted to remove the client implementation at some point, or extract it and put it in another library, because some parts of the code could be made much cleaner if it was possible to start reasoning only in terms of the server part. 

For example, I cannot imagine maintaining also an HTTP 2 client (right now only HTTP 1.1 is supported).

Can you please tell me what kind of features you would like to have built-in? Maybe it's possible to add more features leveraging the built-in support for dependency injection, using the `websockets` library.

**Reply by kamakazikamikaze** [reply]
Thanks for taking the time to reply to my request! I understand that there is a lot to manage already. Moving the HTTP Client to its own repo would be better than completely erasing it since it appears to work pretty well.

I'm just interested in the high-level features, so `read_message`, `write_message`, and setting a compression level. My projects involve distributing work to poller nodes and stream tasks/results over websockets since it's faster than HTTP requests with less bandwidth overhead.
